blocktorrent.py

```diff
@@ -195,10 +193,12 @@ def handle_read(self):
         self.process_message(packet[MAGIC_SIZE:], addr, packet[0:MAGIC_SIZE])
 
     def process_message(self, t, addr, magic):
-        debuglog('btnet', "Received from %s: %s" % (':'.join(map(str, addr)), repr(t)))
+        debuglog('btnet', "Received from %s(%s): %s" % (':'.join(map(str, addr)), magic.encode('hex'), repr(t)))
         peer = None
-        if magic in magic_map:
-            peer = magic_map[magic]
+        if magic in self.magic_map:
+            peer = self.magic_map[magic]
```

Comment:
Using a one-byte magic number as the only way of identifying peers is going to be problematic given that we will likely have at least 50 and possibly more than 255 peers per node. Birthday paradox. We will instead need to check to make sure that the magic number is what we have stored for that IP address, and identify peers based on the address:port their packet comes from (64+16 bits) instead of the macig number they report (8 bits).

Basically, this check should go like this:

if peer in self.peers and self.peers[peer].magic == magic:
    process_message(...)
else: # discard with warning
    debuglog('btnet', "Invalid magic number %s reported by peer %s" % (magic.encode('hex'), peer))